src/client/pages/Forum.js

```diff
@@ -80,6 +19,41 @@ function Forum() {
     document.body.style.overflowY = 'unset';
   }
 
+  function queryPosts() {
+    fetch('http://localhost:5000/api/forum/posts')
+      .then(res => res.json())
+      .then(
+        (serverResult) => {
+          JSON.stringify();
+          setPosts(serverResult);
+          setLoaded(true);
+        }
+      );
+  }
+
+  function handlePostSubmit() {
+    const request = {
+      method: 'POST',
+      mode: 'same-origin',
+      headers: { 'Content-Type': 'application/json' },
+      body: {
+        title: document.getElementById('new-post-field title'),
```

Comment:
Here is a link talking about [Uncontrolled Components](https://reactjs.org/docs/uncontrolled-components.html) and controlled components. Usually, we want to avoid practice like `getElementById` in React component unless we have to. You may want give `createRef` a try, or make the textarea element a controlled one.

Reply:
understood. can't use Refs in functional components, so I went the controlled component route